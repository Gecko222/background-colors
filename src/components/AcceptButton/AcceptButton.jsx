import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { find } from 'lodash';

import './accept-button.css';

/**
 * Accept button
 */
class AcceptButton extends Component {
	/**
	 * render
	 * @return {ReactElement}
	 */
	render() {
		return <button
			className="accept-button"
			onClick={ () => this._onClick() }
		>
			Accept
		</button>;
	}

	/**
	 * on button click
	 */
	_onClick() {
		const { colors, input } = this.props;
		const color = find(colors, color => color.name === input);

		if (color && color.hex) {
			document.body.style.backgroundColor = `#${color.hex}`;
		}
	}

	static propTypes = {
		colors: PropTypes.arrayOf(PropTypes.shape({
			name: PropTypes.string,
			hes: PropTypes.string,
		})),
		input: PropTypes.string,
	}
}

const mapStateToProps = ({ appState }, props) => {
	return {
		colors: appState.colors,
		input: appState.input,
	};
};

export default connect(mapStateToProps)(AcceptButton);
