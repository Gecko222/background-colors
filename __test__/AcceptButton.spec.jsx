import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import AcceptButton from '../src/components/accept-button/accept-button.component';

describe('>>>AcceptButton --- Snapshot', () => {
	const initialState = {
		appState: {
			fetched: false,
			fetchError: false,
			colors: [],
		},
		autocomplete: {},
	};
	const mockStore = configureStore();

	it('+++capturing Snapshot of AcceptButton', () => {
		const store = mockStore(initialState);
		const wrapper = mount( <Provider store={store}><AcceptButton /></Provider> );

		const renderedValue = renderer.create(wrapper).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});

describe('>>>AcceptButton', () => {
	const mockStore = configureStore();

	it('it should change body background color', () => {
		const initialState = {
			appState: {
				fetched: false,
				fetchError: false,
				colors: [{
					name: 'Test',
					hex: 'fff',
				}],
			},
			autocomplete: {
				input: 'Test',
			},
		};

		const store = mockStore(initialState);
		const wrapper = mount( <Provider store={store}><AcceptButton /></Provider> );

		wrapper.find('.accept-button').simulate('click');

		expect(document.getElementsByTagName('body')[0].style.backgroundColor)
		.toBe('rgba(255, 255, 255, 0.5)');
	});

	it('it shouldn\'t change body background color', () => {
		const initialState = {
			appState: {
				fetched: false,
				fetchError: false,
				colors: [{
					name: 'Test',
					hex: '000',
				}],
			},
			autocomplete: {
				input: 'DifrentTest',
			},
		};
		const store = mockStore(initialState);
		const wrapper = mount( <Provider store={store}><AcceptButton /></Provider> );

		wrapper.find('.accept-button').simulate('click');

		expect(document.getElementsByTagName('body')[0].style.backgroundColor)
		.toBe('rgba(255, 255, 255, 0.5)');
	});
});
