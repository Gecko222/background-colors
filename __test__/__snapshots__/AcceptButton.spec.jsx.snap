// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>AcceptButton --- Snapshot +++capturing Snapshot of AcceptButton 1`] = `
<button
  className="accept-button"
  onClick={[Function]}
>
  Accept
</button>
`;
